src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -748,3 +751,69 @@ impl RustcTargetData {
         self.target_config(kind).links_overrides.get(lib_name)
     }
 }
+
+/// Structure used to deal with Rustdoc fingerprinting
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RustDocFingerprint {
+    rustc_vv: String,
+}
+
+impl RustDocFingerprint {
+    /// Read the `RustDocFingerprint` info from the fingerprint file.
+    fn read<'a, 'cfg>(cx: &Context<'a, 'cfg>) -> CargoResult<Self> {
+        let rustdoc_data = paths::read(
+            &cx.files()
+                .host_root()
+                .join(".rustdoc_fingerprint")
+                .with_extension("json"),
+        )?;
+        serde_json::from_str(&rustdoc_data).map_err(|e| anyhow::anyhow!("{:?}", e))
+    }
+
+    /// Write the `RustDocFingerprint` info into the fingerprint file.
+    fn write<'a, 'cfg>(&self, cx: &Context<'a, 'cfg>) -> CargoResult<()> {
+        paths::write(
+            &cx.files()
+                .host_root()
+                .join(".rustdoc_fingerprint.json")
+                .with_extension("json"),
```

Comment:
This doesn't appear to be necessary.

```suggestion
                .with_extension("json"),
```